x-pack/legacy/plugins/siem/public/pages/overview/signals_by_category/index.tsx

```diff
@@ -32,47 +34,45 @@ interface Props {
   to: number;
 }
 
-export const SignalsByCategory = React.memo<Props>(
-  ({
-    deleteQuery,
-    filters = NO_FILTERS,
-    from,
-    query = DEFAULT_QUERY,
-    setAbsoluteRangeDatePicker,
-    setQuery,
-    to,
-  }) => {
-    const updateDateRangeCallback = useCallback(
-      (min: number, max: number) => {
-        setAbsoluteRangeDatePicker!({ id: 'global', from: min, to: max });
-      },
-      [setAbsoluteRangeDatePicker]
-    );
-    const defaultStackByOption = {
-      text: `${i18n.SIGNALS_BY_CATEGORY}`,
-      value: 'signal.rule.threat',
-    };
+const SignalsByCategoryComponent: React.FC<Props> = ({
+  deleteQuery,
+  filters = NO_FILTERS,
+  from,
+  query = DEFAULT_QUERY,
+  setAbsoluteRangeDatePicker,
+  setQuery,
+  to,
+}) => {
+  const { signalIndexName } = useSignalIndex();
+  const updateDateRangeCallback = useCallback(
+    (min: number, max: number) => {
+      setAbsoluteRangeDatePicker!({ id: 'global', from: min, to: max });
+    },
+    [setAbsoluteRangeDatePicker]
+  );
+
+  const defaultStackByOption = signalsHistogramOptions.find(o => o.text === DEFAULT_STACK_BY);
```

Comment:
nit: similar to `EventsByDataset` above, you may want nullish coalescing in case the options change and the default is not found.

e.g.

``` ts
const defaultStackByOption = eventsStackByOptions.find(o => o.text === DEFAULT_STACK_BY) ?? eventsStackByOptions[0];
```